processor/fanoutconnector.go

```diff
@@ -25,6 +26,8 @@ import (
 	"go.opentelemetry.io/collector/internal/data"
 )
 
+const timeout = 5 * time.Second
```

Comment:
The fanoutconnector is used in multiple places (between different components) and we should not have the logic of timeout here.